src/data/nat/gcd/basic.lean

```diff
@@ -6,7 +6,10 @@ Authors: Jeremy Avigad, Leonardo de Moura
 import algebra.group_power.order
 
 /-!
-# Definitions and properties of `gcd`, `lcm`, and `coprime`
+# Definitions and properties of `nat.gcd`, `nat.lcm`, and `nat.coprime`
+
+Generalizations of these are provided in a later file as `gcd_monoid.gcd`, `gcd_monoid.lcm`, and
+`is_coprime`.
```

Comment:
It's a generalization in that on nat they mean the same thing though, right?

Perhaps it's not the expected generalization.... This isn't my area, can you suggest a correction?

Reply:
`nat.coprime` is defeq to the `gcd = 1` condition, and is equivalent to a `is_coprime` condition: [nat.is_coprime_iff_coprime](https://leanprover-community.github.io/mathlib_docs/ring_theory/coprime/lemmas.html#nat.is_coprime_iff_coprime), but coercion to Z is necessary, so I suggest we don't make the connection with `is_coprime` here. (In general, gcd=1 is equivalent to is_coprime for Bezout domains ([is_bezout.gcd_eq_sum](https://leanprover-community.github.io/mathlib_docs/ring_theory/bezout.html#is_bezout.gcd_eq_sum)), in particular PIDs, but `nat` is only a semiring, not a ring.)

Maybe we should instead point to [nat.gcd_monoid](https://leanprover-community.github.io/mathlib_docs/ring_theory/int/basic.html#nat.gcd_monoid) whose definition uses `nat.gcd` and `nat.lcm`, and we probably should add a docstring in the file data.int.gcd as well. I'm also fine with simply not adding this docstring.

Reply:
I agree that the definitions are not literal generalizations, so maybe we can rephrase it as:
```suggestion
The more general equivalents of these are provided in a later file as  `gcd_monoid.gcd`,
`gcd_monoid.lcm`, and `is_coprime`.
```

Reply:
Isn't it weird to say "generalization of [nat.add](https://leanprover-community.github.io/mathlib_docs/init/core.html#nat.add) is provided in a later file as [ring.add](https://leanprover-community.github.io/mathlib_docs/algebra/ring/basic.html#ring)"? It's more common to say that `nat` together with `nat.add` etc. form a `ring`. So here maybe we should just say "`nat` together with `nat.gcd` and `nat.gcm` form a `gcd_monoid`.

Reply:
It's confusing here; I think you probably would say that `+` is a more general spelling of `nat.add`. `gcd` is weird because:

* It has no notation
* It is in the global namespace via `export`, but we have to refer to it as `gcd_monoid.gcd` not `gcd` in order for doc-gen to link it.

Reply:
Okay but maybe just say
```suggestion
Generalizations of these are provided in a later file as `gcd_monoid.gcd` and `gcd_monoid.lcm`.
```

because mentioning `is_coprime` here is inappropriate IMO, since it's not equivalent to gcd=1 in the semiring N (`is_coprime a b` for `a b : N` iff `a=1` or `b=1`). It's more appropriate to mention it in the module docstring of [data.int.gcd](https://leanprover-community.github.io/mathlib_docs/data/int/gcd.html), because the docstring already mentions Bézout's lemma which has the form as the definition of `is_coprime`.

Reply:
Ah, I missed the coercions in `nat.is_coprime_iff_coprime`.